src/commonMain/kotlin/fr/acinq/lightning/payment/IncomingPaymentHandler.kt

```diff
@@ -256,11 +258,13 @@ class IncomingPaymentHandler(val nodeParams: NodeParams, val walletParams: Walle
                                     is PayToOpenPart -> actionForPayToOpenFailure(privateKey, failureMsg, part.payToOpenRequest) // NB: this will fail all parts, we could only return one
                                 }
                             }
+                            payment.parts.filterIsInstance<PayToOpenPart>().firstOrNull()?.let {
```

Comment:
The `firstOrNull()` will select a single payment part, it kind of create the same issue as before. You should use the aggregated amount `payToOpenAmount`, which is what we actually base the decision on. I think the error message should also contain the reference value.

Maybe break up the attributes?

```kotlin
sealed interface PayToOpenEvents : NodeEvents {
    abstract val paymentHash: ByteVector32
    abstract val totalAmount: MilliSatoshi
    abstract val payToOpenAmount: MilliSatoshi
    sealed class Rejected: PayToOpenEvents {
        data class BelowMin(override val paymentHash: ByteVector32, override val totalAmount: MilliSatoshi, override val payToOpenAmount: MilliSatoshi, val minPayToOpenAmount: MilliSatoshi): Rejected()
    }
}
```

Reply:
Fixed in 25039c9